### ISSUE TITLE
[New build - EXPERIMENTAL] 20w32a: Systems Destruction.

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#2746]
This update follows through on the latest effort on combat by making systems destruction a more viable strategy.
It adds a few new gameplay mechanics and some HUD feedback on players own ship and target ship systems HP.

Improvements and changes.
- System Destroy chain: system blocks have a 75% of destroying a neighbor voxel.
- Systems will cease to function under 50% HP.
- HUD: system HP for own and target ship added.
- Thruster system will emit some smoke when partly destroyed.

NB: Player will have ways to improve the previous stats through gameplay in upcoming updates.

Bugfixes:
#2741 Missing children entities on blueprint load.
#2721 #2738 Incorrect chunk information on blueprint load.
#2740 Build on a ship that is selected by the tracking cam breaks game.
#2731 Can link seat to buttons.
#2718 Mag-lock can lock to already locked entity
- Mag-lock can lock circularly.
#2717 Grays paint not applied correctly on light.
#2734 Weapon spread not applied with gimbal weapons.

**Comment by tsunamayo** [comment]
Build is up! Thanks all!

**Comment by Womble-TC** [comment]
In regards to #2717, would you mind painting the purple lights in the first tutorial building back to being purple again? The ones that are just outside of the bedroom. I use the black light buck since it gave off a nice purple colour. But since that got fixed, they are probably just dim grey/black now. Any of the default purple colours should do.

**Comment by MormegiI** [comment]
Don't know if it is just me. But after changing from stable to experimental branch and using new build this happened:
![Floating Buildings](https://user-images.githubusercontent.com/51251532/89215071-41a92100-d5c8-11ea-9a71-85cda8f51ddb.PNG)

**Comment by tsunamayo** [comment]
@MormegiI Yes this is a regression, it is fixed I will do a hotfix patch soon. For the TIE I will also need the blueprint of course! Thanks

**Comment by tsunamayo** [comment]
@Womble-TC oh I see, let me check this out. One other thing we might have to change soon is the number of lights in the landing pad, with the upcoming light tech it will likely wont look good and will be a bit perf intensive. Cheers

**Comment by MormegiI** [comment]
@tsunamayo Here are 3 Blueprints affected by the issue. I guess all my old BPs are but they should be enough for you to find out what's wrong. In case they can't be fixed just tell me and i'll rebuild. Need to rework them anyway now that i can use brick-systems.

[Blueprints.zip](https://github.com/tsunamayo/Starship-EVO/files/5022708/Blueprints.zip)

**Comment by tsunamayo** [comment]
@MormegiI I dont see any issues, are you sure the children entity where not build on the legacy hinge? Thanks

**Comment by Kaiser-Indrasil** [comment]
I love the high chance of systems blowing up neighbouring blocks!

There is something weird about how quickly ships can be destroyed in this patch. I mean, I hardly dealt any damage and it's already dead.
![image](https://user-images.githubusercontent.com/63756057/89363033-2709a180-d6c7-11ea-8a03-f1cf76e58099.png)

**Comment by MormegiI** [comment]
@tsunamayo i think not but i'm actually not sure any more. Realized there are some BPs missing the the folder as well so they may be on my other computer. Doesn't matter too much anyway, i'll just rebuild.

**Comment by tsunamayo** [comment]
@Kaiser-Indrasil not sure, I didnt really changed anything regarding HP durability. But the system chain explosion can definitely destroy adjacent hull blocks. If you have some decent ship (system-wise I mean, not look wise), I would like to see them, I am testing a lot combat these days!

**Comment by Kaiser-Indrasil** [comment]
I only shot the ones piloted by the NPCs which are available in-game. I've noticed they go down way sooner than before the patch, concerningly so.
